base256emoji.go

```diff
@@ -0,0 +1,130 @@
+package multibase
+
+import "strconv"
+
+var base256emojiTable = [256]rune{
+	// Curated list, this is just a list of things that *somwhat* are related to our comunitty
+	'🚀', '🙬', '🙭', '🙮', '🙯', '🪐', '☄', '🛰', '🌌', // Space
+	'🌑', '🌒', '🌓', '🌔', '🌕', '🌖', '🌗', '🌘', // Moon
+	'🌍', '🌏', '🌎', // Our Home, for now (earth)
+	'☉', '☼', '☀', // Our Garden, for now (sol)
+	'💻', '🖥', '🖳', '💾', '💿', // Computer
+	// The rest is completed from https://home.unicode.org/emoji/emoji-frequency/ at the time of creation (december 2021) (the data is from 2019), most used first until we reach 256.
+	// We exclude modifier based emojies (such as flags) as they are bigger than one single codepoint.
+	'😂', '❤', '😍', '🤣', '😊', '🙏', '💕', '😭', '😘', '👍',
+	'😅', '👏', '😁', '♥', '🔥', '🥰', '💔', '💖', '💙', '😢',
+	'🤔', '😆', '🙄', '💪', '😉', '☺', '👌', '🤗', '💜', '😔',
+	'😎', '😇', '🌹', '🤦', '🎉', '‼', '💞', '✌', '✨', '🤷',
+	'😱', '😌', '🌸', '🙌', '😋', '💗', '💚', '😏', '💛', '🙂',
+	'💓', '🤩', '😄', '😀', '🖤', '😃', '💯', '🙈', '👇', '🎶',
+	'😒', '🤭', '❣', '❗', '😜', '💋', '👀', '😪', '😑', '💥',
+	'🙋', '😞', '😩', '😡', '🤪', '👊', '🥳', '😥', '🤤', '👉',
+	'💃', '😳', '✋', '😚', '😝', '😴', '🌟', '😬', '🙃', '🍀',
+	'🌷', '😻', '😓', '⭐', '✅', '🥺', '🌈', '😈', '🤘', '💦',
+	'✔', '😣', '🏃', '💐', '☹', '🎊', '💘', '😠', '☝', '😕',
+	'🌺', '🎂', '🌻', '😐', '🖕', '💝', '🙊', '😹', '🗣', '💫',
+	'💀', '👑', '🎵', '🤞', '😛', '🔴', '😤', '🌼', '😫', '⚽',
+	'🤙', '☕', '🏆', '🧡', '🎁', '⚡', '🌞', '🎈', '❌', '✊',
+	'👋', '😲', '🌿', '🤫', '👈', '😮', '🙆', '🍻', '🍃', '🐶',
+	'💁', '😰', '🤨', '😶', '🤝', '🚶', '💰', '🍓', '💢', '🤟',
+	'🙁', '🚨', '💨', '🤬', '✈', '🎀', '🍺', '🤓', '😙', '💟',
+	'🌱', '😖', '👶', '🥴', '▶', '➡', '❓', '💎', '💸', '⬇',
+	'😨', '🌚', '🦋', '😷', '🕺', '⚠', '🙅', '😟', '😵', '👎',
+	'🤲', '🤠', '🤧', '📌', '🔵', '💅', '🧐', '🐾', '🍒', '😗',
+	'🤑', '🌊', '🤯', '🐷', '☎', '💧', '😯', '💆', '👆', '🎤',
+	'🙇', '🍑', '❄', '🌴', '💣', '🐸', '💌', '📍', '🥀', '🤢',
+	'👅', '💡', '💩', '⁉', '👐', '📸', '👻', '🤐',
+}
+
+var base256emojiReverseTable = map[rune]byte{
```

Comment:
Rather than have the copy/paste from this array and "base256emojiTable" above, can we just derive this array from base256emojiTable ?

Reply:
I know it's not a thing you said but still, this look very error prone but this is 100% checked by the `TestBase256EmojiTableMatches` test.

The goal was since:
- There is no writes only reads to this map.
- This is private (so no extra surprising usage after compilation).

It is then trivial for a compiler to see that this is read only constants.
I was hopping `gc` to compile that to a static `.data` segment and avoid bloating the `init` even more.
However after checking, what happen in practice is that `gc` allocates a `uintptr` in the `.bss`, and heap allocate the map at `init` which is worst than what you propose.

I'll make that change.